core/lls_core/models/lattice_data.py

```diff
@@ -74,6 +75,21 @@ def read_image(cls, values: dict):
         # Use the Deskew version of this validator, to do the actual image loading
         return super().read_image(values)
 
+    @validator("input_image", pre=True, always=True)
+    def incomplete_final_frame(cls, v: DataArray) -> Any:
+        """
+        Check final frame, if acquisition is stopped halfway through it causes failures
+        This validator will remove a bad final frame
+        """
+        final_frame = v[-1]

```

Comment:
Note that `v[-1]` will just take the last index along the first axis, but not the other axes. Is this what you want here?
```python
>>> arr
<xarray.DataArray (T: 2, C: 3, Z: 4, Y: 5, X: 6)>
>>> arr[-1]
<xarray.DataArray (C: 3, Z: 4, Y: 5, X: 6)>
```

Reply:
Yep that's correct. I only want to check the final frame, that's the only one that should fail in this way, I could probably drop to a single channel as well to save processing time but it's a pretty marginal improvement and this is clearer I think

Reply:
In that case, we could reduce the memory usage and possibly clarify what is happening by doing:
```suggestion
        final_frame = v.isel(T=-1, C=-1)
```

Reply:
Yeah that is a bit more explicit, will make that change

Reply:
You can click the "commit suggestion" button to use the change directly if you want.